### PR TITLE
*TEMPORARY* US251839 tab style updates

### PR DESCRIPTION
**Pull request by starryeyez024**
This is to see the diff against the local vars branch, original PR is here:
https://github.com/patternfly/patternfly-elements/pull/948

### CODE REVIEW

File: elements/pfe-tabs/src/pfe-tab-panel.scss [hunk 1]
```diff
@@ -17,12 +17,21 @@ $LOCAL-VARIABLES: (
...
   }
-
   [tabindex] {
     display: flex;
     height: 100%;
   }
 
+ //updated saturated styles
+  @at-root #{&}([pfe-variant="earth"][on="saturated"]) .container{
+    background-color: #fff;
```
Comment (starryeyez024) [inline elements/pfe-tabs/src/pfe-tab-panel.scss:27]: These could be converted to theme colors
```suggestion
    background-color: pfe-var(surface--lightest);
```

File: elements/pfe-tabs/src/pfe-tab-panel.scss [hunk 2]
```diff
@@ -17,12 +17,21 @@ $LOCAL-VARIABLES: (
...
+    color: #000 !important;
```
Comment (starryeyez024) [inline elements/pfe-tabs/src/pfe-tab-panel.scss:28]:
```suggestion
    color: pfe-var(text);
```

File: elements/pfe-tabs/src/pfe-tab.scss [hunk 3]
```diff
@@ -150,49 +156,177 @@ $LOCAL-VARIABLES: (
...
+  }
+}
+
+/// ===========================================================================
+/// WIND TAB VARIANTS / saturated
+/// ===========================================================================
+
+:host {
+  @at-root #{&}([on="saturated"][pfe-variant="wind"]) {
+    color: #{pfe-var(link--focus--on-dark)};
```
Comment (starryeyez024) [inline elements/pfe-tabs/src/pfe-tab.scss:246]:
```suggestion
    color: #{pfe-var(text--focus--on-saturated)};
```

File: elements/pfe-tabs/src/pfe-tab.scss [hunk 4]
```diff
@@ -150,49 +156,177 @@ $LOCAL-VARIABLES: (
...
+    }
+  }
+  @at-root #{&}([on="saturated"][aria-selected="false"][pfe-variant="wind"]:hover) {
+    --pfe-tabs__tab--BorderBottom: #{pfe-local(BorderWidth--hover)} #{pfe-var(surface--border-style)} #{pfe-local(BorderColor--hover)};
+    --pfe-tabs__tab--BorderBottom: 0 !important;
+    --pfe-tabs__tab--BorderBottom: #{pfe-local(BorderWidth--hover)} #{pfe-var(surface--border-style)} #{pfe-local(BorderColor--hover)};
+    --pfe-tabs__tab--BorderBottom: #{pfe-local(BorderWidth--hover)} #{pfe-var(surface--border-style)} #{pfe-var(surface--lightest)} !important;
+  }
+  @at-root #{&}([on="saturated"][aria-selected="true"][pfe-variant="wind"]) {
+    color: var(--pfe-theme--color--surface--lightest, #fff) !important;
```
Comment (starryeyez024) [inline elements/pfe-tabs/src/pfe-tab.scss:259]:
```suggestion
    color: pfe-var(text--on-saturated) !important;
```

File: elements/pfe-tabs/src/pfe-tab-panel.scss (hunk 2 above)
Comment (starryeyez024) [reply elements/pfe-tabs/src/pfe-tab-panel.scss:28]: I'm using the "text" color variable instead of "link" throughout these suggestions because the mockups treat the copy as if it is regular text instead of a link (there's no blue). Basically there is enough other design treatment around the copy to make it stand out as a UI element, thus it doesn't have to lean on the bright blue color to let users know it's clickable.

File: elements/pfe-tabs/src/pfe-tab.scss [hunk 5]
```diff
@@ -38,17 +41,17 @@ $LOCAL-VARIABLES: (
 
   // Inner style
   background-color: #{pfe-local(BackgroundColor)};
-  color: #{pfe-local(Color)};
+  color: #{pfe-var(ui-disabled--text)};  // @todo replace with pfe(text-muted) in Kendall's color story
```
Comment (starryeyez024) [inline elements/pfe-tabs/src/pfe-tab.scss:44]:
```suggestion
  color: pfe-local(Color);
```

File: elements/pfe-tabs/src/pfe-tab.scss [hunk 6]
```diff
@@ -3,17 +3,20 @@
 $LOCAL: tabs;
 
 $LOCAL-VARIABLES: (
-  BackgroundColor: transparent,
-  Color:           pfe-var(ui-disabled--text),
-  link:            pfe-var(link),
-  focus:           pfe-var(link--focus),
-  highlight:       pfe-var(ui-accent),
+  BackgroundColor:       transparent,
+  Color:                 pfe-var(ui-disabled--text),
```
Comment (starryeyez024) [inline elements/pfe-tabs/src/pfe-tab.scss:7]:
```suggestion
  Color:                 pfe-var(text--muted),
```

File: elements/pfe-tabs/src/pfe-tab.scss [hunk 7]
```diff
@@ -38,17 +41,17 @@ $LOCAL-VARIABLES: (
...
+  text-align: #{pfe-local(TextAlign, left, $region: tab)};
+
   // Exposing this for customization
   text-transform: #{pfe-local($cssvar: TextTransform, $region: tab, $fallback: none)};
   font-family:  #{pfe-var(font-family)};
   font-weight:  #{pfe-var(font-weight--normal)};
   font-size:    #{pfe-local(FontSize, $region: tab)};
-    
+
   @for $i from 1 through 6 {
```
Comment (starryeyez024) [inline elements/pfe-tabs/src/pfe-tab.scss:55]: its not part of this PR, but can we fix / remove this:
https://www.evernote.com/l/AA4w1aXMJwpGnZ_Rqf521LpFd9uWVsTkpDg

we can just remove the wrapper `  @for $i from 1 through 6 {	` and its closing bracket

File: elements/pfe-tabs/src/pfe-tab.scss [hunk 8]
```diff
@@ -98,42 +98,48 @@ $LOCAL-VARIABLES: (
...
     }
   }
 }
 
 :host([pfe-variant="earth"]) {
   @include browser-query(ie11) {
-    background-color: #{pfe-var(surface--lighter)};
-    color: #{pfe-var(text)};
+    background-color: #{pfe-fetch(surface--lighter)};
+    color:            #{pfe-fetch(ui-disabled--text)}; // @todo replace with pfe(text-muted) in Kendall's color story
```
Comment (starryeyez024) [inline elements/pfe-tabs/src/pfe-tab.scss:134]: you can go ahead and add the text-muted vars to the `_colors.scss` file in this branch:

```
  text:                        $pfe-color--text,
  text--muted:                 $pfe-color--gray-600,
  link:                        $pfe-color--link,
  link--visited:               $pfe-color--link--visited,
  link--hover:                 $pfe-color--link--hover,
  link--focus:                 $pfe-color--link--focus,

  text--on-dark:               $pfe-color--text--on-dark,
  text--muted--on-dark:        $pfe-color--gray-50,
  link--on-dark:               $pfe-color--link--on-dark,
  link--visited--on-dark:      $pfe-color--link--visited--on-dark,
  link--hover--on-dark:        $pfe-color--link--hover--on-dark,
  link--focus--on-dark:        $pfe-color--link--focus--on-dark,

  text--on-saturated:          $pfe-color--text--on-saturated,
  text--muted--on-saturated:   $pfe-color--gray-50,
  link--on-saturated:          $pfe-color--link--on-saturated,
  link--visited--on-saturated: $pfe-color--link--visited--on-saturated,
  link--hover--on-saturated:   $pfe-color--link--hover--on-saturated,
  link--focus--on-saturated:   $pfe-color--link--focus--on-saturated,
```

File: elements/pfe-tabs/src/pfe-tab.scss [hunk 9]
```diff
@@ -150,49 +156,177 @@ $LOCAL-VARIABLES: (
...
 /// ===========================================================================
 
 :host([pfe-variant="earth"][aria-selected="false"]) {
-  --pfe-tabs--BackgroundColor: #{pfe-var(surface--lighter)};
-
-  --pfe-tabs__tab--BorderLeft:   #{pfe-var(ui--border-width)} #{pfe-var(ui--border-style)} #{pfe-var(surface--border)};
-  --pfe-tabs__tab--BorderBottom: #{pfe-var(ui--border-width)} #{pfe-var(ui--border-style)} #{pfe-var(surface--border)};
-  --pfe-tabs__tab--BorderTop:    #{pfe-var(surface--border-width--heavy)} #{pfe-var(ui--border-style)} #{pfe-var(surface--lighter)};
+  $variantEarth: (
+    BackgroundColor: #f0f0f0,
```
Comment (starryeyez024) [inline elements/pfe-tabs/src/pfe-tab.scss:285]:
```suggestion
    BackgroundColor: pfe-var(surface--lighter),
```

**Comment by starryeyez024** [comment]
@alazzara  Please add this to the demo.css file:
```css
.saturated-background {
    background-color: teal;
    --theme: saturated;
}
```
And update the demo.html file to include saturated examples:
http://pastebin.test.redhat.com/884717  (line 351 - 405)

File: elements/pfe-tabs/src/pfe-tab.scss (hunk 3 above)
Comment (alazzara) [reply elements/pfe-tabs/src/pfe-tab.scss:246]: @starryeyez024 This gave me a dark grey color and was hard to read on saturated BG's. I used `ui-base--hover--on-saturated` instead. Let me know if that works.

Comment (starryeyez024) [reply elements/pfe-tabs/src/pfe-tab.scss]: ah my bad, this was a combination of me first suggesting we use the --on-saturated version of the link var, but then i went back and changed to to "text" which doesn't have a focus state! 

You'll likely want to use one of these:
`text--on-saturated` or `text--muted--on-saturated`

File: elements/pfe-icon/demo/pfe-tab.css [hunk 10]
```diff
@@ -0,0 +1,461 @@
...
+    }
+  }
+}
+
+/// ===========================================================================
+/// WIND TAB VARIANTS / saturated
+/// ===========================================================================
+:host {
+  @at-root #{&}([on="saturated"][pfe-variant="wind"]) {
+    color: #{pfe-var(link--focus--on-dark)};
```
Comment (starryeyez024) [inline elements/pfe-icon/demo/pfe-tab.css:251]: @castastrophe do we want to override the value of the `--pfe-tabs--Color` local variable here, instead of redefining the color property?

File: elements/pfe-icon/demo/pfe-tab.css [hunk 11]
```diff
@@ -0,0 +1,461 @@
...
+  }
+}
+
+/// ===========================================================================
+/// EARTH TAB VARIANTS
+/// ===========================================================================
+
+:host([pfe-variant="earth"][aria-selected="false"]) {
+  $variantEarth: (
+    BackgroundColor: #f0f0f0,
```
Comment (starryeyez024) [inline elements/pfe-icon/demo/pfe-tab.css:270]: same suggestion here, should this override `--pfe-tabs--BackgroundColor` and set the value to `pfe-var(surface--lighter)`
@castastrophe  @alazzara

File: elements/pfe-icon/demo/pfe-tab.css (hunk 10 above)
Comment (castastrophe) [reply elements/pfe-icon/demo/pfe-tab.css:251]: I think in general we actually want to move in this direction - redefining the property, `color`, for example, instead of updating the local variable, such as `--pfe-tabs--Color`.  This would allow `--pfe-tabs--Color` to remain empty in the component so that the variable hook continues to work, even if attributes are used.

File: elements/pfe-tabs/src/pfe-tab.scss [hunk 12]
```diff
@@ -3,17 +3,21 @@
...
 $LOCAL-VARIABLES: (
-  BackgroundColor: transparent,
-  Color:           pfe-var(ui-disabled--text),
-  link:            pfe-var(link),
-  focus:           pfe-var(link--focus),
-  highlight:       pfe-var(ui-accent),
+  BackgroundColor:       transparent,
+  Color:                 pfe-var(text--muted),
+  link:                  pfe-var(link),
+  text-hover:            #151515,
```
Comment (castastrophe) [inline elements/pfe-tabs/src/pfe-tab.scss:9]:
```suggestion
  text--hover:            #151515,
```

Modifiers are separated by double dashes in BEM

File: elements/pfe-tabs/src/pfe-tab.scss [hunk 13]
```diff
@@ -3,17 +3,21 @@
...
+  BorderColor--hover:    pfe-var(surface--base),
+  darkBackgroundColor:   #3c3f42,
   tab: (
-    FontSize:       pfe-var(font-size),
-    PaddingTop:     calc(#{pfe-var(container-padding)} * .666),
-    PaddingRight:   calc(#{pfe-var(container-padding)} * 1.5),
-    PaddingBottom:  calc(#{pfe-var(container-padding)} * .666),
-    PaddingLeft:    calc(#{pfe-var(container-padding)} * 1.5),
+    FontSize:            pfe-var(font-size),
+    PaddingTop:          calc(#{pfe-var(container-padding)}),
```
Comment (castastrophe) [inline elements/pfe-tabs/src/pfe-tab.scss:17]:
```suggestion
    PaddingTop:          pfe-var(container-padding),
```
No calc necessary here

File: elements/pfe-tabs/src/pfe-tab.scss [hunk 14]
```diff
@@ -3,17 +3,21 @@
...
-  link:            pfe-var(link),
-  focus:           pfe-var(link--focus),
-  highlight:       pfe-var(ui-accent),
+  BackgroundColor:       transparent,
+  Color:                 pfe-var(text--muted),
+  link:                  pfe-var(link),
+  text-hover:            #151515,
+  focus:                 pfe-var(link--focus),
+  highlight:             pfe-var(ui-accent),
+  BorderWidth--hover:    #{pfe-var(surface--border-width--active)},
```
Comment (castastrophe) [inline elements/pfe-tabs/src/pfe-tab.scss:12]:
```suggestion
  BorderWidth--hover:    pfe-var(surface--border-width--active),
```

No interpolation needed

File: elements/pfe-tabs/src/pfe-tab.scss [hunk 15]
```diff
@@ -3,17 +3,21 @@
...
+  BorderColor--hover:    pfe-var(surface--base),
+  darkBackgroundColor:   #3c3f42,
```
Comment (castastrophe) [inline elements/pfe-tabs/src/pfe-tab.scss:14]:
```suggestion
BackgroundColor--on-dark:   #3c3f42,
```

File: elements/pfe-icon/demo/pfe-tab.css (hunk 11 above)
Comment (castastrophe) [reply elements/pfe-icon/demo/pfe-tab.css:270]: @starryeyez024 That is what's being done here - it's defined in the map and the `pfe-print-local` mixin prints out all the custom properties for you.

File: elements/pfe-icon/demo/pfe-tab.css (hunk 10 above)
Comment (starryeyez024) [reply elements/pfe-icon/demo/pfe-tab.css:251]: I totes see the value of that, but how do we decide how to do one vs the other? looking at the temp pfe-cta.css file in the master branch, I see the various attributes redefining the local variables:
image

Comment (castastrophe) [reply elements/pfe-icon/demo/pfe-tab.css:251]: That's kind of what I'm saying. We're not doing this now (redefining the properties) but maybe we should be.
